gettsim/benefits/elterngeld.py

```diff
@@ -1,232 +1,245 @@
-        },
+    """Calculate elterngeld given the relevant wage and the eligibility for bonuses."""
+    alternative_elterngeld = (
+        elterngeld_eink_erlass.clip(
+            lower=elterngeld_params["elterngeld_mindestbetrag"],
+            upper=elterngeld_params["elterngeld_höchstbetrag"],
+        )
+        + geschw_bonus
+        + anz_mehrlinge_bonus
     )
 
-    household.loc[~household["elternzeit_anspruch"], ["elterngeld_m"]] = 0
+    data = np.where(
+        (elterngeld_eink_relev < 0) | ~elternzeit_anspruch, 0, alternative_elterngeld
+    )
 
-    return household
+    return pd.Series(index=elterngeld_eink_relev.index, data=data)
 
 
-def calc_elterngeld(
-    person,
-    params,
-    soz_vers_beitr_params,
-    eink_st_abzuege_params,
+def proxy_eink_vorj_elterngeld(
+    beitr_bemess_grenze_rentenv,
+    bruttolohn_vorj_m,
+    elterngeld_params,
     eink_st_params,
+    eink_st_abzuege_params,
     soli_st_params,
 ):
-    """ Calculating elterngeld given the relevant wage and the eligibility on sibling
-    bonus.
+    """Calculating the claim for benefits depending on previous wage.
+
+    TODO: This function requires `.fillna(0)` at the end. Investigate!
 
     """
-    considered_wage = calc_considered_wage(
-        person,
-        params,
-        soz_vers_beitr_params,
-        eink_st_abzuege_params,
+    # Relevant wage is capped at the contribution thresholds
+    max_wage = bruttolohn_vorj_m.clip(upper=beitr_bemess_grenze_rentenv)
+
+    # We need to deduct lump-sum amounts for contributions, taxes and soli
+    prox_ssc = elterngeld_params["elterngeld_soz_vers_pausch"] * max_wage
+
+    # Fictive taxes (Lohnsteuer) are approximated by applying the wage to the tax tariff
+    prox_tax = _st_tarif(
+        12 * max_wage - eink_st_abzuege_params["werbungskostenpauschale"],
         eink_st_params,
-        soli_st_params,
     )
 
-    if considered_wage < 0:
-        person["elterngeld_m"] = 0
-    else:
-        paid_percentage = calc_elterngeld_percentage(considered_wage, params)
+    prox_soli = piecewise_polynomial(
+        prox_tax,
+        thresholds=soli_st_params["soli_st"]["thresholds"],
+        rates=soli_st_params["soli_st"]["rates"],
+        intercepts_at_lower_thresholds=soli_st_params["soli_st"][
+            "intercepts_at_lower_thresholds"
+        ],
+    )
 
-        elterngeld_calc = considered_wage * paid_percentage
+    return (
+        (max_wage - prox_ssc - prox_tax / 12 - prox_soli / 12).clip(lower=0).fillna(0)
+    )
 
-        prelim_elterngeld = max(
-            min(elterngeld_calc, params["elterngeld_höchstbetrag"]),
-            params["elterngeld_mindestbetrag"],
-        )
-        if person["geschw_bonus"]:
-            prelim_elterngeld += calc_geschw_bonus(elterngeld_calc, params)
-        prelim_elterngeld += (
-            person["anz_mehrlinge_bonus"] * params["elterngeld_mehrling_bonus"]
+
+def date_of_birth(geburtsjahr, geburtsmonat, geburtstag):
```

Comment:
We should collect all variables which are not thematically connected in an extra module. Also: kids in hh/tu, adults in hh/tu etc.